fhir2dataset/parser.py

```diff
@@ -152,31 +130,64 @@ def __init_clauses(self):
                 "WHERE": self.__where_parser,
             }
         )
-        for clause, function in BASIC_CLAUSES.items():
-            self.CLAUSES[f" {clause} "] = function
-            self.CLAUSES[f"{clause} "] = function
-            self.CLAUSES[f" {clause}"] = function
-            self.CLAUSES[clause] = function
 
     def __init_split_mask(self):
-        self.__split_mask_clauses = self.__create_mask_multiple_separators(self.CLAUSES.keys())
-        self.__split_mask_from = " AS "
-        self.__split_mask_select = " , |, "
-        self.__split_mask_from_join = " ON "
-        self.__split_mask_join = " = "
-        self.__split_mask_where = " AND "
-        self.__split_mask_where_condition = self.__create_mask_multiple_separators(PREFIX + [" = "])
-
-    def __create_mask_multiple_separators(self, separators, keep_separators: bool = True):
-        split_mask_multiple = ""
-        for separator in separators:
-            if split_mask_multiple:
-                split_mask_multiple += f"|{separator}"
-            else:
-                split_mask_multiple += f"{separator}"
+        self.__split_mask_clauses = self.__create_mask(
+            self.CLAUSES.keys(), can_start_sentence=True,
+        )
+
+        self.__split_mask_from = self.__create_mask("AS", keep_separators=False,)
+
+        self.__split_mask_select = self.__create_mask(
+            ",", keep_separators=False, optional_space_before=True
+        )
+
+        self.__split_mask_from_join = self.__create_mask("ON", keep_separators=False,)
+
+        self.__split_mask_join = self.__create_mask("=", keep_separators=False,)
+
+        self.__split_mask_where = self.__create_mask("AND", keep_separators=False,)
+
+        self.__split_mask_where_condition = self.__create_mask(PREFIX + ["="])
+
+    def __create_mask(
+        self,
+        separators,
+        keep_separators=True,
```

Comment:
Much better!!
Can we set the default keep_separators to True to simplify the calls above?

Reply:
It's changed :+1: